.github/workflows/CI.yml

```diff
@@ -51,3 +54,43 @@ jobs:
         flags: unittests
         name: codecov-umbrella
         fail_ci_if_error: false
+
+
+  deploy:
+    name: Deploy
+    needs: [tests]
+    runs-on: ubuntu-latest
+    if: github.ref=='refs/heads/main' && github.event_name!='pull_request'
+
+    permissions:
+      contents: write
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
```

Comment:
suggestion: might be worth going up to 3.12

Reply:
The choice of Python 3.10 was actually quite intentional. Since all this is used for is deployment, there isn't any benefit to installing the latest version of Python, and there can sometimes be downsides until the latest version has stabilized. I imagine Python 3.11 would work just fine, but since I'd have to check and see whether there are other existing workflows that use that version without issue, I prefer to [choose boring technology](https://mcfunley.com/choose-boring-technology) and use Python 3.10 until I know for certain that later versions work without problems. 😊

Reply:
Fine with me. Thx for the explanation! 👌